### DISCUSSION TITLE
Http接口返回重复记录

### DISCUSSION THREAD

**Discussion by luluo9527** [hiroi-sora/Umi-OCR#505] [General]
大佬，我根据我们业务要求使用Python语言开发了一个图像批量转换工具，在工具中试图调用您开发的Umi-OCR 2.1 软件的http接口，将多幅JPG图像批量OCR识别为TXT文件。我们使用此工具调用 http [接口时，发现程序能顺利识别文字，但是返回的结果中出现了多条重复的文字记录，附件是我保存的http接口返回的json文件，请您协助看看这是什么原因，谢谢
[results.json.zip](https://github.com/hiroi-sora/Umi-OCR/files/15220960/results.json.zip)

**Comment by hiroi-sora** [comment]
你好。

1. 你上传的 results.json ，并不是Umi-OCR直接返回的内容格式，可能是你们自己转换后的格式，因此我无法判断问题所在。Umi-OCR自身应该不会出现重复返回结果的情况；大概率是您解析OCR结果时出了问题。
2. 如果你们在本地进行OCR（无需借助局域网服务器），那么直接用Python调用 [PaddleOCR-json](https://github.com/hiroi-sora/PaddleOCR-json) 会更高效，相当于Umi-OCR去除网络层。
3. 如果你们检查/调整代码后仍有重复的情况，可以将关键代码部分发上来，我可以帮忙看看。

**Comment by luluo9527** [comment]
你好大佬，您的邮件已收到，我再按您建议检查下代码，稍后把结果反馈回来，非常感谢您的及时回复！


                       
Original Email
                       
                     

Sender:"hiroi-sora"< ***@***.*** &gt;;

Sent Time:2024/5/6 21:20

To:"hiroi-sora/Umi-OCR"< ***@***.*** &gt;;

Cc recipient:"luluo9527"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

Subject:Re: [hiroi-sora/Umi-OCR] Http接口返回重复记录 (Discussion #505)




 
你好。
  
你上传的 results.json ，并不是Umi-OCR直接返回的内容格式，可能是你们自己转换后的格式，因此我无法判断问题所在。Umi-OCR自身应该不会出现重复返回结果的情况；大概率是您解析OCR结果时出了问题。
 
如果你们在本地进行OCR（无需借助局域网服务器），那么直接用Python调用 PaddleOCR-json 会更高效，相当于Umi-OCR去除网络层。
 
如果你们检查/调整代码后仍有重复的情况，可以将关键代码部分发上来，我可以帮忙看看。
